packages/grafana-runtime/src/services/LocationService.tsx

```diff
@@ -40,6 +43,12 @@ export class HistoryWrapper implements LocationService {
         ? H.createMemoryHistory({ initialEntries: ['/'] })
         : H.createBrowserHistory({ basename: config.appSubUrl ?? '/' }));
 
+    this.locationObservable = new BehaviorSubject(this.history.location);
+
+    this.history.listen((location) => {
```

Comment:
Seems like this is the same way react-router listens for changes to update it's router context but we need it too because we need to react on location changes outside of router context.